src/server/journal/journal_slice.cc

```diff
@@ -68,20 +75,24 @@ void JournalSlice::SetFlushMode(bool allow_flush) {
 }
 
 void JournalSlice::AddLogRecord(const Entry& entry) {
-  DCHECK(ring_buffer_);
+  DCHECK(ring_buffer_.capacity() > 0);
+
+  ring_buffer_.push_back(JournalItem());
```

Comment:
I think we need to add statistics for this buffer size which we will expose in metrics. we can do that in a different PR